### PR TITLE
Add basic support for buildah in releaseApplication

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**

### CODE REVIEW

**Review by hamza3202 — CHANGES_REQUESTED** [review]
We need to disable dockerconfig in case of buildah

**Comment by ahmedwaleedmalik** [comment]
> We need to disable dockerconfig in case of buildah

@hamza3202 For authentication buildah looks first for the default auth file at **$HOME/containers/auth.json** which is set using `buildah login` and if it cannot find it there then it looks at **$HOME/.docker/config.json** which is set using `docker login`. So dockerconfig might serve us well in that regard. But yes authentication is something that needs to be looked at.

**Comment by hamza3202** [comment]
Great! But we need to atleast not mount the docker socket for buildah.

**Comment by ahmedwaleedmalik** [comment]
@hamza3202 agreed. I'll add that

File: src/io/stakater/pods/Pod.groovy [hunk 1]
```diff
@@ -6,17 +6,21 @@ def setToolsImage(Map parameters = [:], String image) {
...
-
 Map setDockerConfig(Map parameters = [:]) {
-    mountDockerSocket(parameters)
+    setPodVolumes(parameters)
     setDefaultContainerEnvVarsConfig(parameters)
 
-    parameters.podVolumes.isDockerConfig = true
-    parameters.podContainers.defaultContainer.envVarsConfig.isDocker = true
+    // Don't mount docker socket if useBuildah is true or isDockerMount is true
+    if (!((parameters.containsKey("useBuildah") && parameters.get('useBuildah', true )) ||
```
Comment (hamza3202) [inline src/io/stakater/pods/Pod.groovy]: `useBuildah` shouldn't affect the mounting of docker socket. Only `isDockerMount` should be enough. Also the comment is wrong.
If `isDockerMount` is true then docker socket should be mounted and that should be the only condition.

File: src/io/stakater/pods/Pod.groovy [hunk 2]
```diff
@@ -6,17 +6,21 @@ def setToolsImage(Map parameters = [:], String image) {
...
+            parameters.containsKey("isDockerMount") && parameters.get('isDockerMount', false ))) {
+        parameters.podVolumes.isDockerMount = true
+        parameters.podContainers.defaultContainer.envVarsConfig.isDocker = true
+    }
+
+    // Don't mount docker config if isDockerConfig is false
+    if (!(parameters.containsKey("isDockerConfig") && parameters.get('isDockerConfig', false ))) {
```
Comment (hamza3202) [inline src/io/stakater/pods/Pod.groovy]: Does the condition need to be inverted here? Seems like if the key doesn't exist or if value of `parameters. isDockerConfig` is false then `parameters.podVolumes.isDockerConfig = true` would be set to true?

File: src/io/stakater/pods/Pod.groovy (hunk 1 above)
Comment (hamza3202) [reply src/io/stakater/pods/Pod.groovy]: `useBuildah` could instead set `parameters.isDockerMount` to `false` if needed.

**Review by hamza3202 — CHANGES_REQUESTED** [review]

File: src/io/stakater/pods/Pod.groovy (hunk 2 above)
Comment (ahmedwaleedmalik) [reply src/io/stakater/pods/Pod.groovy]: If key exists and key is set to false, then isDockerConfig will use the default value false. Else isDockerConfig should be set to true.

File: src/io/stakater/pods/Pod.groovy (hunk 1 above)
Comment (ahmedwaleedmalik) [reply src/io/stakater/pods/Pod.groovy]: Handled

File: src/io/stakater/pods/Pod.groovy (hunk 2 above)
Comment (ahmedwaleedmalik) [reply src/io/stakater/pods/Pod.groovy]: Handled

**Review by hamza3202 — APPROVED** [review]

**Review by stakater-user — APPROVED** [review]

**Pull request merged**